src/account/operations/transfer/high_level/burn_native_token.rs

```diff
@@ -0,0 +1,156 @@
+}
+
+impl AccountHandle {
+    fn contains_bech32_address(address: &str, account_addresses: &[AccountAddress]) -> crate::Result<Address> {
+        let (_bech32_hrp, address) = Address::try_from_bech32(address)?;
+        if account_addresses
+            .binary_search_by_key(&address, |address| address.address.inner)
+            .is_err()
+        {
+            return Err(Error::AddressNotFoundInAccount);
+        }
+        Ok(address)
+    }
+
+    /// Function to burn native tokens with foundry
+    pub async fn burn_native_token(
+        &self,
+        burn_native_token_options: BurnNativeTokenOptions,
+        options: Option<TransferOptions>,
+    ) -> crate::Result<TransferResult> {
+        log::debug!("[TRANSFER] burn_native_token");
+        let byte_cost_config = self.client.get_byte_cost_config().await?;
+
+        let account_addresses = self.list_addresses().await?;
+        // the address needs to be from the account, because for the minting we need to sign transactions from it
+        let controller_address = match &burn_native_token_options.account_address {
+            Some(address) => AccountHandle::contains_bech32_address(address, &account_addresses)?,
+            None => {
+                account_addresses
+                    .first()
+                    // todo other error message
+                    .ok_or(Error::FailedToGetRemainder)?
+                    .address
+                    .inner
+            }
+        };
+
+        let token_id = burn_native_token_options.native_token.0;
+        let burn_token_amount = burn_native_token_options.native_token.1;
+
+        let foundry_id = token_id.foundry_id();
+        let alias_id = *foundry_id.alias_address().alias_id();
+
+        let account = self.read().await;
+
+        let mut existing_alias_output_data = None;
+        let mut existing_foundry_output = None;
+        account
+            .unspent_outputs()
+            .values()
+            .into_iter()
+            .filter(|output_data| match &output_data.output {
+                Output::Alias(output) => output.alias_id().or_from_output_id(output_data.output_id) == alias_id,
+                Output::Foundry(output) => output.id() == foundry_id,
+                _ => false,
+            })
+            .for_each(|output_data| match &output_data.output {
+                Output::Alias(_) => existing_alias_output_data = Some(output_data),
+                Output::Foundry(output) => existing_foundry_output = Some(output),
+                _ => unreachable!("We checked if it's an alias or foundry output before"),
+            });
+
+        let existing_alias_output_data = existing_alias_output_data
+            .ok_or_else(|| Error::BurningFailed("No alias output available".to_string()))?
+            .clone();
+
+        let existing_foundry_output = existing_foundry_output
+            .ok_or_else(|| Error::BurningFailed("No foundry output available".to_string()))?
```

Comment:
Just to be more clear it shouldn't be `No` but `Required` or something similar, because we could have other alias or foundry outputs available, but they can just not be used to burn the provided native tokens

Reply:
I don't quite understand your point here, can you explain this again? Thanks

Reply:
`No alias output available` sounds like the wallet has no Alias output, but it could have another one, that is unrelated to the token id

Reply:
Oh I get, I'll correct the error message. Thank you